cylc/flow/workflow_files.py

```diff
@@ -1048,22 +1050,7 @@ def get_platforms_from_db(run_dir):
         pri_dao.close()
 
 
-@overload
-def parse_reg(reg: str, src: Literal[False] = False) -> str:
-    ...
-
-
-@overload
-def parse_reg(reg: str, src: Literal[True]) -> Tuple[str, Path]:
-    ...
-
-
-@overload
-def parse_reg(reg: str, src: bool) -> Union[str, Tuple[str, Path]]:
-    ...  # Need this 3rd overload https://github.com/python/mypy/issues/6113
-
-
-def parse_reg(reg: str, src: bool = False) -> Union[str, Tuple[str, Path]]:
+def parse_reg(reg: str, src: bool = False) -> Tuple[str, Path]:
```

Comment:
@MetRonnie - not sure if you'll try to talk me out of this. The mypy overloading stuff is cool, but it seems overly complicated to me, for all its worth in this case, especially as mypy was giving me hell trying to adapt this for Cylc 7 back-compat on this branch.  It's more straightforward to have `parse_reg()` always return `(name, path)` and have callers just ignore one element or the other as needed.  What do think?

Reply:
Perhaps we should just split it up? Something like `parse_reg(reg: str) -> str` and `parse_src(workflow: str) -> Tuple[str, Path]`. Not sure why I didn't do that in the first place, it seems simple enough unless I've missed something

Reply:
Yeah we could do that, as a follow-up.